torch/onnx/_internal/exporter/_compat.py

```diff
@@ -117,6 +118,9 @@ def export_compat(
                 # so we reverse the list to maintain the order of the custom ops provided
                 registry.register_op(torch_op, op, is_complex=False)
     try:
+        assert GLOBALS.in_onnx_export is False
+        GLOBALS.in_onnx_export = True
+
         onnx_program = _core.export(
```

Comment:
Should we set the flag in _core? In fact, we could set a different flag in core, then check it in the is_in_onnx_export function. We can also lift the is_in_onnx_export  function from utils to `__init__` to show that it is no longer an old exporter thing.

https://github.com/pytorch/pytorch/blob/1d3c50fcc5156e6509d2a64727401c01a5998df0/torch/onnx/__init__.py#L71

Reply:
If there isn't a try catch in _core.export, we can create a decorator that has

Reply:
Done